src/System.Text.Json/src/System/Text/Json/Document/JsonDocument.Parse.cs

```diff
@@ -29,19 +29,26 @@ public sealed partial class JsonDocument
         ///   </para>
         /// </remarks>
         /// <param name="utf8Json">JSON text to parse.</param>
-        /// <param name="readerOptions">Options to control the reader behavior during parsing.</param>
+        /// <param name="options">Options to control the reader behavior during parsing.</param>
         /// <returns>
         ///   A JsonDocument representation of the JSON value.
         /// </returns>
         /// <exception cref="JsonException">
         ///   <paramref name="utf8Json"/> does not represent a valid single JSON value.
         /// </exception>
         /// <exception cref="ArgumentException">
-        ///   <paramref name="readerOptions"/> contains unsupported options.
+        ///   <paramref name="options"/> contains unsupported options.
         /// </exception>
-        public static JsonDocument Parse(ReadOnlyMemory<byte> utf8Json, JsonReaderOptions readerOptions = default)
+        public static JsonDocument Parse(ReadOnlyMemory<byte> utf8Json, JsonDocumentOptions options = default)
         {
-            CheckSupportedOptions(readerOptions);
+            CheckSupportedOptions(options);
+
+            var readerOptions = new JsonReaderOptions
+            {
+                AllowTrailingCommas = options.AllowTrailingCommas,
+                CommentHandling = options.CommentHandling,
+                MaxDepth = options.MaxDepth
+            };
```

Comment:
Rather than duplicating this code anywhere it's needed, how about having an internal ToReaderOptions on JsonDocumentOptions?